datalake/src/test/java/com/sequenceiq/datalake/service/sdx/stop/SdxStopServiceTest.java

```diff
@@ -78,18 +89,40 @@ public class SdxStopServiceTest {
     @Mock
     private RegionAwareInternalCrnGenerator regionAwareInternalCrnGenerator;
 
+    @Mock
+    private FlowChainLogService flowChainLogService;
+
+    @Mock
+    private FlowLogService flowLogService;
+
     @InjectMocks
     private SdxStopService underTest;
 
     @Test
     public void testTriggerStop() {
+        when(flowLogService.getLastFlowLog(anyLong())).thenReturn(Optional.empty());
         SdxCluster sdxCluster = sdxCluster();
-
         underTest.triggerStopIfClusterNotStopped(sdxCluster);
-
         verify(sdxReactorFlowManager).triggerSdxStopFlow(sdxCluster);
     }
 
+    @Test
+    public void testTriggerStopWhenUnstoppableFlowRunning() {
+        FlowLog flowLog = new FlowLog();
+        when(flowLogService.getLastFlowLog(anyLong())).thenReturn(Optional.of(flowLog));
+
+        SdxCluster sdxCluster = sdxCluster();
+        for (String flowChainType : UNSTOPPABLE_FLOWS) {
+            when(flowChainLogService.getFlowChainType(any())).thenReturn(flowChainType);
+            try {
+                underTest.triggerStopIfClusterNotStopped(sdxCluster);
+                fail("Expected exception due to " + flowChainType + " being an unstoppable flow.");
+            } catch (CloudbreakServiceException e) {
+                assertEquals(UNSTOPPABLE_FLOW_EXCEPTION_MESSAGE, e.getMessage());
+            }
```

Comment:
Can you please use `Assertions.assertThrows` istead of the `try-catch`?

Reply:
Will do.